gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/BundleMapperService.java

```diff
@@ -136,6 +140,16 @@ public Bundle mapToBundle(RCMRIN030000UK06Message xmlMessage) {
         return bundle;
     }
 
+    private void mapDiagnosticReports(Bundle bundle, RCMRMT030101UK04EhrExtract ehrExtract, Patient patient, List<Encounter> encounters,
+        List<Observation> observations, List<Observation> observationComments, String practiceCode) {
+        var diagnosticReports = diagnosticReportMapper.mapDiagnosticReports(ehrExtract, patient, encounters, practiceCode);
+        var specimen = diagnosticReportMapper.mapSpecimen(ehrExtract, diagnosticReports, patient, practiceCode);
+        addEntries(bundle, diagnosticReports);
+        addEntries(bundle, specimen);
+        diagnosticReportMapper.mapChildObservationComments(ehrExtract, observationComments);
+        diagnosticReportMapper.addSpecimenChildReferences(ehrExtract, observations, observationComments, diagnosticReports);
```

Comment:
Assuming this works, but should we not be only adding these entries once we've finished with the mapChildObservationComments and addSpecimenChildReferences?